clippy_lints/src/methods/mod.rs

```diff
@@ -1796,6 +1797,30 @@ declare_clippy_lint! {
     "replace `.splitn(2, pat)` with `.split_once(pat)`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Check for `&mut iter` followed by a method call.
+    ///
+    /// ### Why is this bad?
+    /// This requires using parenthesis to signify precedence.
```

Comment:
Let's be explicit here.

```suggestion
    /// Using `(&mut iter)._` requires using parenthesis to signify precedence.
```